### PR TITLE
themix-gui: init at 1.15.1

### LINKED ISSUE #151564: package-request: oomox
**Project description**
Graphical application for generating different color variations of a Materia and Oomox themes (GTK2, GTK3 and others), Archdroid, Gnome-Colors, Numix, Papirus, Suru++ icons, and terminal palette.

**Metadata**

* homepage URL: https://github.com/themix-project/oomox
* source URL: https://github.com/themix-project/oomox
* license: gpl3
* platforms: linux

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

Adds themix-gui, including oomox-gui; project page at https://github.com/themix-project/themix-gui.
Fixes #151564.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/th/themix-gui/package.nix [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+, fetchFromGitHub
+, wrapGAppsHook
+}:
+
+let
+  py = python3.withPackages (p: [
+    p.pygobject3
+  ]);
+in
+stdenv.mkDerivation rec {
```
Comment (Lurkki14) [inline pkgs/by-name/th/themix-gui/package.nix:19]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

Comment (Lurkki14) [reply pkgs/by-name/th/themix-gui/package.nix:19]: Recursive attributes in `stdenv.mkDerivation` should be done with the `finalAttrs` pattern as described here https://nixos.org/manual/nixpkgs/stable/#mkderivation-recursive-attributes

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/th/themix-gui/package.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/by-name/th/themix-gui/package.nix:19]: Yeah everyone has their own opinion on this one. It's less relevant here as it's only used for the fetcher and not in the actual build, i.e. fetcher has to be overridden regardless. Howevere, using `finalAttrs` introduces an eval time penalty that many argue is not worth it.

Comment (eclairevoyant) [reply pkgs/by-name/th/themix-gui/package.nix:19]: I can convert it to a `let`-expr if that reads better.

**Review by rvolosatovs — APPROVED** [review]

File: pkgs/by-name/th/themix-gui/package.nix [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  ];
+
+  buildInputs = [
+    gdk-pixbuf
+    glib
+    gtk3
+    librsvg
+  ];
+
+  propagatedBuildInputs = [
```
Comment (pbsds) [inline pkgs/by-name/th/themix-gui/package.nix:46]:
```suggestion
```

File: pkgs/by-name/th/themix-gui/package.nix [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    py
+  ];
+
+  postPatch = ''
+    substituteInPlace packaging/bin/{oomox,themix}-gui --replace python3 ${lib.getExe py}
```
Comment (pbsds) [inline pkgs/by-name/th/themix-gui/package.nix:51]:
```suggestion
    substituteInPlace gui.sh packaging/bin/{oomox,themix}-gui --replace python3 ${lib.getExe py}
```

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 261960` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>themix-gui</li>
  </ul>
</details>

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

Runs fine, just a small improvement then LGTM

**Comment by eclairevoyant** [comment]
Going through the readme again, this appears to be Linux-specific, so I've set platforms accordingly.

**Comment by noggynoggy** [comment]
Hey there!

I was trying to use the themix-gui package (thank you by the way!), but out of the box it seems there aren't any of the necessary plugins installed (like the oomox or material themes or any of the icon plugins). Causing the GUI to be basically useless(?).
I tried to hack in the oomox plugin going of the AUR package themix-theme-oomox-git, but as it turns out: Skill Issue. 

Is anyone using the package? If so, how do you use it? What are the steps to have it working?

**Pull request merged**